test/e2e/specs/widgets/customizing-widgets.spec.js

```diff
@@ -463,9 +466,9 @@ test.describe( 'Widgets Customizer', () => {
 		await page.keyboard.press( 'Escape' );
 		await expect(
 			page.locator(
-				'*[aria-live="polite"][aria-relevant="additions text"] >> text=/^You are currently in navigation mode./'
+				'css=.block-editor-block-list__layout.is-navigate-mode'
```

Comment:
Could we find another accessible selector for this? Using CSS selectors is generally not recommended.

Reply:
I tried, but this was the only indication in the HTML markup that the editor is in navigation mode. Previously this test used the "you are currently in navigation mode" text in the `aria-live` region used by `speak()`, but that gets quickly overwritten by the next `speak()` call.

Reply:
Some ideas:

1. If the `speak()` call gets overridden, does it also mean that the message might not be accessible to screen readers or other assistive technologies? Will that be a concern in this case?
2. A common fallback to CSS selectors is using `data-testid` (or other public API if there's any), is it suitable here?

Anyway, this isn't that important and more like a nitpick 😅. We might want to restrict CSS selectors in e2e tests in the long term though, it'd be good to start these discussions early 🙂 .

Reply:
> If the `speak()` call gets overridden, does it also mean that the message might not be accessible to screen readers or other assistive technologies?

The problem here is that when you press <kbd>Escape</kbd> to enter the navigation mode, two things happen at once:
1. The editor mode changes, and is announced by saying "you are currently in navigation mode".
2. The `BlockSelectionButton` for the selected block appears (it's the black thing on the screenshot below) and it announces itself by saying "Paragraph Block. Row 1."

<img width="357" alt="Screenshot 2023-02-03 at 14 52 42" src="https://user-images.githubusercontent.com/664258/216621508-28353ce9-b80f-45d3-a556-5b654babf645.png">

Both happen at the same time, while handling the <kbd>Escape</kbd> key event. The order is determined by random things like which store listeners runs first or which update is scheduled to which microtask. Enabling concurrent mode reversed this order.

What does the screen reader do in this case? I don't know. Does it say the first message and then the second? Or does it ignore the first one, because it disappeared very quickly?